src/main/java/hng_java_boilerplate/region/controller/RegionController.java

```diff
@@ -0,0 +1,240 @@
+            @ApiResponse(responseCode = "401", description = "Unauthorized request", content = @Content(schema = @Schema(implementation = RegionErrorResponseDto.class))),
+            @ApiResponse(responseCode = "409", description = "Conflict, region already assigned to the user", content = @Content(schema = @Schema(implementation = RegionErrorResponseDto.class)))
+    })
+    @PostMapping()
+    public ResponseEntity<?> assignRegion(
+            @RequestBody UserRegionDto userRegionDTO,
+            Authentication authentication
+    ){
+        try {
+        //check if region is predefined
+        if(!regionService.isRegionAvailable(userRegionDTO.getRegionName())){
+            return new ResponseEntity<>(
+                    Optional.of(RegionErrorResponseDto.builder()
+                            .statusCode("404")
+                            .message("Not found")
+                            .build()),
+                    HttpStatus.BAD_REQUEST
+            );
+        }
+
+        // Ensure that assignee is owner of account.
+        User user = (User) authentication.getPrincipal();
+        String user_id = user.getId();
+        if(!user_id.equals(userRegionDTO.getUserId())){
+            return new ResponseEntity<>(
+                    Optional.of(RegionErrorResponseDto.builder()
+                            .statusCode("401")
+                            .message("Unauthorized request")
+                            .build()),
+                    HttpStatus.UNAUTHORIZED
+            );
+        }
+            if(regionService.hasUserAssignedRegion(userRegionDTO.getUserId())){
+                return new ResponseEntity<>(
+                        Optional.of(RegionErrorResponseDto.builder()
+                                .statusCode("409")
+                                .message("Conflict")
+                                .build()),
+                        HttpStatus.CONFLICT
+                );
+            }
+            UserRegionEntity userRegionEntity = userRegionMapper.mapFrom(userRegionDTO);
+            userRegionEntity.setRegionCode(userRegionDTO.getCountryCode());
+            Optional<?> assignedRegion = regionService.saveUserRegion(userRegionEntity);
+            UserRegionDto region = userRegionMapper.mapTo((UserRegionEntity) assignedRegion.get());
+            return new ResponseEntity<>(
+                    UserRegionResponseDto.builder()
+                            .regionName(region.getRegionName())
+                            .countryCode(region.getCountryCode())
+                            .userId(region.getUserId())
+                            .build()
+                    , HttpStatus.OK);
+        } catch (Exception e) {
+            System.out.println("========= Error message =========> " + e.getMessage());
```

Comment:
You can use logger here instead of printlin. There is a lombok dependeny on the application. annotate your class with slf4j and you get the logger. Better for prod code.